tests/flakes.sh

```diff
@@ -714,5 +723,264 @@ git -C $flakeB commit -a -m 'Foo'
 
 [[ $(nix eval --update-input b $flakeA#foo) = 1912 ]]
 
+# Test overrides
+
+# Test simple override
+# A(override B.C=C2) -> B -> C
+cat > $flakeA/flake.nix <<EOF
+{
+  inputs.b.url = git+file://$flakeB;
+  inputs.b.inputs.c.url = git+file://$flakeC2;
+  outputs = { self, b }: {
+    foo = 1 + b.foo;
+  };
+}
+EOF
+git -C $flakeA add flake.nix
+git -C $flakeA commit -a -m 'initial'
+
+cat > $flakeB/flake.nix <<EOF
+{
+  inputs.c.url = git+file://$flakeC;
+  outputs = { self, c }: {
+    foo = 1 + c.foo;
+  };
+}
+EOF
+git -C $flakeB add flake.nix
+git -C $flakeB commit -a -m 'initial'
+
+cat > $flakeC/flake.nix <<EOF
+{
+  outputs = { self }: {
+    foo = 1;
+  };
+}
+EOF
+git -C $flakeC add flake.nix
+git -C $flakeC commit -a -m 'initial'
+
+cat > $flakeC2/flake.nix <<EOF
+{
+  outputs = { self }: {
+    foo = 2;
+  };
+}
+EOF
+git -C $flakeC2 add flake.nix
+git -C $flakeC2 commit -a -m 'initial'
+
+# B + C = 2
+[[ $(nix eval $flakeB#foo --recreate-lock-file) = 2 ]]
+# A + B + C2 = 4
+[[ $(nix eval $flakeA#foo --recreate-lock-file) = 4 ]]
+
+# Test simple override from command line
+# A -> B -> C
+# --input-override a.b.c = C2
+cat > $flakeA/flake.nix <<EOF
+{
+  inputs.b.url = git+file://$flakeB;
+  outputs = { self, b }: {
+    foo = 1 + b.foo;
+  };
+}
+EOF
+git -C $flakeA add flake.nix
+git -C $flakeA commit -a -m 'initial'
+
+# B, C, and C2 same as previous test
+# A + B + C = 3
+[[ $(nix eval $flakeA#foo --recreate-lock-file) = 3 ]]
+# A + B + C2 = 4
+[[ $(nix eval $flakeA#foo --recreate-lock-file --override-input b/c git+file://$flakeC2) = 4 ]]
+
+# Test unused overrides detected
+# A(override B.X=A, B.X.X=B, B.Y=C, B.C.Z=D) -> B -> C
+cat > $flakeA/flake.nix <<EOF
+{
+  inputs.b.url = git+file://$flakeB;
+  inputs.b.inputs.x.url = git+file://$flakeA;
+  inputs.b.inputs.x.inputs.x.url = git+file://$flakeB;
+  inputs.b.inputs.y.url = git+file://$flakeC;
+  inputs.b.inputs.c.inputs.z.url = git+file://$flakeD;
+  outputs = { self, b }: {
+    foo = 1 + b.foo;
+  };
+}
+EOF
+git -C $flakeA add flake.nix
+git -C $flakeA commit -a -m 'initial'
+
+# B and C same as previous test
+[[ $(nix eval $flakeA#foo --recreate-lock-file) = 3 ]]
+expectedOutputRegex="warning:[[:space:]]unused[[:space:]]override\(s\):[[:space:]]b/c/z=git\+file:///.*flakeD
```

Comment:
I would leave out the `[[:space:]]` stuff, it makes the test harder to maintain (since it's much more likely that we would change the text of the warning than the whitespace).